src/verifier.rs

```diff
@@ -13,30 +12,29 @@ use lightning_block_sync::rest::RestClient;
 use crate::config;
 
 pub(crate) struct ChainVerifier {
-	rest_client: Arc<RestClient>,
+	rest_client: RestClient,
 }
 
 struct RestBinaryResponse(Vec<u8>);
 
 impl ChainVerifier {
 	pub(crate) fn new() -> Self {
-		let rest_client = RestClient::new(config::bitcoin_rest_endpoint()).unwrap();
 		ChainVerifier {
-			rest_client: Arc::new(rest_client),
+			rest_client: RestClient::new(config::bitcoin_rest_endpoint()).unwrap(),
 		}
 	}
 
 	fn retrieve_block(&self, block_height: u32) -> Result<Block, AccessError> {
-		let rest_client = self.rest_client.clone();
 		tokio::task::block_in_place(move || { tokio::runtime::Handle::current().block_on(async move {
-			let block_hash_result = rest_client.request_resource::<BinaryResponse, RestBinaryResponse>(&format!("blockhashbyheight/{}.bin", block_height)).await;
+			let block_hash_result = self.rest_client.request_resource::<BinaryResponse, RestBinaryResponse>
+				(&format!("blockhashbyheight/{}.bin", block_height)).await;
```

Comment:
honestly, extract the format! into a separate variable so this, too, looks less busy